pkg/operators/ebpf/ebpf.go

```diff
@@ -593,8 +594,13 @@ func (i *ebpfInstance) Start(gadgetCtx operators.GadgetContext) error {
 			if t == nil {
 				continue
 			}
-			i.logger.Debugf("replacing map %q", v.name)
-			mapReplacements[v.name] = t
+			if err := i.collectionSpec.Maps[v.name].Compatible(t); err == nil {
+				mapReplacements[v.name] = t
+			} else {
+				i.logger.Warnf("Replacement map %q incompatible: %s",
```

Comment:
It's more difficult because at this place in the code, the ebpf operator doesn't know about the socket enricher. It just knows about the variable with the ebpf map to replace.

I think the variable type should be changed to a struct that contains the purpose of the map replacement, so the error message can tell about the feature that would not work.